lib/MetricsConsumer.js

```diff
@@ -0,0 +1,90 @@
+'use strict'; // eslint-disable-line strict
+
+const Logger = require('werelogs').Logger;
+
+const { RedisClient, StatsClient } = require('arsenal').metrics;
+const BackbeatConsumer = require('./BackbeatConsumer');
+const redisKeys = require('../extensions/replication/constants').redisKeys;
+
+// StatsClient constant defaults
+const INTERVAL = 300; // 5 minutes;
+const EXPIRY = 900; // 15 minutes
+
+// BackbeatConsumer constant defaults
+const CONSUMER_FETCH_MAX_BYTES = 5000020;
+const CONCURRENCY = 10;
+
+class MetricsConsumer {
+    /**
+     * @constructor
+     * @param {object} rConfig - redis configurations
+     * @param {string} rConfig.host - redis host
+     * @param {number} rConfig.port - redis port
+     * @param {object} mConfig - metrics configurations
+     * @param {string} mConfig.topic - metrics topic name
+     * @param {object} zkConfig - zookeeper configurations
+     * @param {string} zkConfig.connectionString - zookeeper connection string
+     *   as "host:port[/chroot]"
+     */
+    constructor(rConfig, mConfig, zkConfig) {
+        this.mConfig = mConfig;
+        this.zkConfig = zkConfig;
+
```

Comment:
Could use validation of the config values to protect against anything unexpected.

Reply:
I feel like validating within the constructor seems a little too much because it basically would mean the wrong params were passed to the class. This could translate to any other class and method out there. I see classes like `BackbeatProducer` using joi validation, but it looks like it's mainly because all params are passed together within the same object.